rest-api/src/inference.py

```diff
@@ -164,6 +164,48 @@ def infer_from_text(
         except:
             traceback.print_exc()
             return np.zeros(1)
+        
+    def preprocess_text(self,
+        input_text: str,
+        lang: str,
+        # speaker_name: str,
+        transliterate_roman_to_native: bool = True
+    ) -> np.ndarray:
+        try:
+            
+            # If there's no separate English model, use the Hinglish one if present
+            if lang == "en" and lang not in self.models and "en+hi" in self.models:
+                lang = "en+hi"
+
+            if lang == "en+hi": # Hinglish (English+Hindi code-mixed)
+                primary_lang, secondary_lang = lang.split('+')
+                # TODO: Write a proper `transliterate_native_words_using_eng_dictionary`
+            else:
+                primary_lang = lang
+                secondary_lang = None
+
+            input_text = self.text_normalizer.normalize_text(input_text, primary_lang)
+            if secondary_lang:
+                input_text = self.transliterate_native_words_using_spell_checker(input_text, secondary_lang)
+
+            if transliterate_roman_to_native and primary_lang != 'en':
+                input_text = self.transliterate_sentence(input_text, primary_lang)
+
+                # Manipuri was trained using the Central-govt's Bangla script
+                # So convert the words in native state-govt script to Eastern-Nagari
+                if primary_lang == "mni":
+                    # TODO: Delete explicit-schwa
+                    input_text = aksharamukha_xlit("MeeteiMayek", "Bengali", input_text)
+            return input_text
+        except Exception:
+            traceback.print_exc()
+            return ""
```

Comment:
Can you please remove this exception-handling? Praveen had put this earlier, but this seems to cause issues on Dhruva, because we are unable to detect Triton errors and log them to understand what input caused this issue.

Please also do the same in `infer_from_text`

Reply:
For example, there is an issue raised which says that the TTS API returns empty audio for the following inputs:

**Assamese:**
```
প্ৰাৰম্ভিক শৈশৱ শিক্ষাৰ বাবে ৰাষ্ট্ৰীয় পাঠ্যক্ৰম ফ্ৰেমৱৰ্ক (ইচিচিই) হৈছে ভাৰতৰ ইচিচিই সেৱা প্ৰদানকাৰীসকলৰ বাবে এক পথ প্ৰদৰ্শক দস্তাবেজ। ই প্ৰাৰম্ভিক শৈশৱৰ শিক্ষাত মানদণ্ড আৰু উৎকৃষ্টতাৰ প্ৰচাৰৰ বাবে শিশুৰ যত্ন আৰু প্ৰাৰম্ভিক শৈক্ষিক অনুশীলনৰ বাবে নিৰ্দেশনা প্ৰদান কৰে। পৰিকাঠামোটোক তিনিটা খণ্ডত ভাগ কৰা হৈছেঃ প্ৰাৰম্ভিক যত্ন আৰু শিক্ষণৰ ভেটি, প্ৰাৰম্ভিক যত্ন আৰু শিক্ষণৰ লক্ষ্য, আৰু কাৰ্যসূচী পৰিকল্পনা আৰু অনুশীলন। ইয়াত ই. চি. চি. ই.-ৰ তাত্ত্বিক ভেটি, শৈশৱৰ প্ৰাৰম্ভিক শিক্ষাৰ উদ্দেশ্য, বিকাশৰ ক্ষেত্ৰ, শিক্ষামূলক দৃষ্টিভংগী, কাৰ্যসূচী পৰিকল্পনা, মূল্যায়ন, আৰু যত্ন লোৱাসকল/ই. চি. চি. ই. শিক্ষকসকলৰ ভূমিকা আদি বিষয়সমূহ সামৰি লোৱা হৈছে। এই পৰিকাঠামোৰ উদ্দেশ্য হৈছে প্ৰাৰম্ভিক বছৰৰ পেছাদাৰী, সেৱা প্ৰদানকাৰী, ইচিচিই শিক্ষক/যত্ন লোৱাসকল, সম্প্ৰদায় আৰু ৰাজ্য চৰকাৰসমূহক জন্মৰ পৰা প্ৰাক-প্ৰাথমিক বছৰলৈকে শিশুসকলৰ বাবে সমৃদ্ধ প্ৰাৰম্ভিক উদ্দীপনা আৰু শিক্ষণৰ অভিজ্ঞতা প্ৰদান কৰাত সহায় কৰা। ইয়াত ৰাষ্ট্ৰীয় ইচিচিই পাঠ্যক্ৰম ফ্ৰেমৱৰ্কৰ এটা লিংক দিয়া হৈছেঃ
```

Oriya:
```
ପ୍ରାରମ୍ଭିକ ଶୈଶବ ଶିକ୍ଷା ପାଇଁ ଜାତୀଯ଼ ପାଠ୍ଯ଼କ୍ରମ ଫ୍ରେମୱାର୍କ (ଇସିସିଇ) ହେଉଛି ଭାରତର ସମସ୍ତ ଅଞ୍ଚଳରେ ଇସିସିଇ ସେବା ପ୍ରଦାନକାରୀଙ୍କ ପାଇଁ ଏକ ମାର୍ଗଦର୍ଶକ ଦସ୍ତାବିଜ। ଶିଶୁ ଯତ୍ନ ଏବଂ ପ୍ରାରମ୍ଭିକ ଶିକ୍ଷାଗତ ଅଭ୍ଯ଼ାସ ପାଇଁ ନିର୍ଦ୍ଦେଶାବଳୀ ପ୍ରଦାନ କରି ପ୍ରାରମ୍ଭିକ ବାଲ୍ଯ଼ ଯତ୍ନ ଏବଂ ଶିକ୍ଷାରେ ଗୁଣବତ୍ତା ଏବଂ ଉତ୍କର୍ଷତାକୁ ପ୍ରୋତ୍ସାହିତ କରିବା ଏହାର ଲକ୍ଷ୍ଯ଼ | ପ୍ରାରମ୍ଭିକ ବର୍ଷର ପେସାଦାର, ସେବା ପ୍ରଦାନକାରୀ, ଇସିସିଇ ଶିକ୍ଷକ/ଯତ୍ନ ନେଉଥିବା ବ୍ଯ଼କ୍ତି, ଗୋଷ୍ଠୀ ଏବଂ ରାଜ୍ଯ଼ ସରକାରମାନଙ୍କୁ ଜନ୍ମରୁ ପ୍ରାକ୍-ପ୍ରାଥମିକ ବର୍ଷ ପର୍ଯ୍ଯ଼ନ୍ତ ପିଲାମାନଙ୍କ ପାଇଁ ସମୃଦ୍ଧ ପ୍ରାରମ୍ଭିକ ଉଦ୍ଦୀପନା ଏବଂ ଶିକ୍ଷଣ ଅଭିଜ୍ଞତା ପ୍ରଦାନ କରିବାରେ ସହାଯ଼ତା କରିବା ଏହି ଢାଞ୍ଚାର ଉଦ୍ଦେଶ୍ଯ଼। ଏହି ଢାଞ୍ଚାକୁ ତିନୋଟି ବିଭାଗରେ ବିଭକ୍ତ କରାଯାଇଛିଃ ପ୍ରାରମ୍ଭିକ ଯତ୍ନ ଏବଂ ଶିକ୍ଷଣର ମୂଳଦୁଆ, ପ୍ରାରମ୍ଭିକ ଯତ୍ନ ଏବଂ ଶିକ୍ଷଣର ଲକ୍ଷ୍ଯ଼, ଏବଂ କାର୍ଯ୍ଯ଼କ୍ରମ ଯୋଜନା ଏବଂ ଅଭ୍ଯ଼ାସ। ଫାଉଣ୍ଡେସନ ବିଭାଗରେ ଏକ ପରିଚଯ଼, ଏକ ଭାରତୀଯ଼ ଶିଶୁ ପାଇଁ ଦୂରଦୃଷ୍ଟି, ଯୁକ୍ତି ଏବଂ ଇସିସିଇ ପାଇଁ ତତ୍ତ୍ୱଗତ ମୂଳଦୁଆ ଅନ୍ତର୍ଭୁକ୍ତ | ଛଅ ବର୍ଷରୁ କମ୍ ବଯ଼ସର ପିଲାମାନଙ୍କର ସାମଗ୍ରିକ ବିକାଶ ସୁନିଶ୍ଚିତ କରିବା ପାଇଁ ବିକାଶର ବିଭିନ୍ନ କ୍ଷେତ୍ର, ଅର୍ଥାତ୍ ଶାରୀରିକ, ଭାଷା, ଜ୍ଞାନାତ୍ମକ, ସାମାଜିକ-ଭାବପ୍ରବଣ, ଏବଂ ସୃଜନଶୀଳ ଏବଂ ସୌନ୍ଦର୍ଯ୍ଯ଼ ପ୍ରଶଂସାକୁ ପ୍ରୋତ୍ସାହିତ କରିବା ପାଇଁ ଲକ୍ଷ୍ଯ଼ ବିଭାଗ ଅନ୍ତର୍ଭୁକ୍ତ | କାର୍ଯ୍ଯ଼କ୍ରମ ଯୋଜନା ଏବଂ ଅଭ୍ଯ଼ାସ ବିଭାଗ କାର୍ଯ୍ଯ଼କାରିତା ବିବରଣୀ ଉପରେ ଧ୍ଯ଼ାନ ଦେଇଥାଏ ଯେପରିକି କାର୍ଯ୍ଯ଼କ୍ରମ ଯୋଜନାର ନୀତି, କାର୍ଯ୍ଯ଼କ୍ରମ ଯୋଜନାର ପର୍ଯ୍ଯ଼ାଯ଼, ପିତାମାତା ଏବଂ ଯତ୍ନ ନେଉଥିବା ବ୍ଯ଼କ୍ତି/ଇସିସିଇ ଶିକ୍ଷକଙ୍କ ଭୂମିକା, ଅତ୍ଯ଼ାବଶ୍ଯ଼କ ଖେଳ ସାମଗ୍ରୀ ଏବଂ ମୂଲ୍ଯ଼ାଙ୍କନ ପ୍ରକ୍ରିଯ଼ା ଇତ୍ଯ଼ାଦି। ଏଠାରେ ଜାତୀଯ଼ ଇସିସିଇ ପାଠ୍ଯ଼କ୍ରମ ଢାଞ୍ଚାର ଏକ ଲିଙ୍କ ଦିଆଯାଇଛିଃ https://wcd.nic.in/sites/default/files/ECCE%20Curriculum%20Framework%20%26%20Quality%20Standards.pdf
```

Because of the above exception handling, we are unable to detect from Dhruva why this is happening.

Reply:
I'm getting the error on Dhruva
Seems an UTF8 char issue. 2 possible things
1. It's having some issue and printing the value out. Triton is using ascii terminal by default hence `ordinal not in range(128)` error. Just modifying the lib to not print solves this
2. Need to figure out why this byte is not recognised. Quick fix would give incomplete output